scripts/process_speech_commands_data.py

```diff
@@ -314,17 +339,30 @@ def __process_data(data_folder: str, dst_folder: str, rebalance: bool = False, c
                 fout.write('\n')
                 fout.flush()
 
+                file_count += 1
+                if file_count % pct_file == 0:
+                    if not skip_duration:
+                        logging.info(f"Finished serializing {file_count} / {num_files} into {manifest_filename}")
+
         logging.info(f"Finished construction of manifest : {manifest_filename}")
 
+        if skip_duration:
+            logging.info(
+                f"\n<<NOTE>> Duration computation was skipped for demonstration purposes on Colaboratory.\n"
+                f"In order to replicate paper results and properly perform data augmentation, \n"
+                f"please recompute the manifest file without the `--skip_duration` flag !\n"
+            )
+
 
 def main():
     parser = argparse.ArgumentParser(description='Google Speech Command Data download')
     parser.add_argument("--data_root", required=True, default=None, type=str)
     parser.add_argument('--data_version', required=True, default=1, type=int, choices=[1, 2])
     parser.add_argument('--class_split', required=False, default='all', type=str, choices=['all', 'sub'])
     parser.add_argument('--rebalance', required=False, action='store_true')
+    parser.add_argument('--skip_duration', required=False, action='store_true')
     parser.add_argument('--log', required=False, action='store_true')
-    parser.set_defaults(log=False, rebalance=False)
+    parser.set_defaults(log=False, rebalance=False, skip_duration=False)
```

Comment:
Isn't this line redundant since they are "store_true" anyways.

Reply:
That seems to be the way argparse doc suggests using these action flags. It avoids a case where the flag is none cause no value is set and no flag is passed.

Reply:
Do you have the link to where this is recommended? I'm never heard of it done this way. Would be nice to know if this is the recommended way.

Reply:
So trying to find that doc, it seems I need to brush up on py3.x argparse. The above behaviour is in 2.7, and 3.2+ auto sets the default as the inverse of store_*. We can remove it safely. It's still used in various code bases where the flag itself is unclear as to what the default value is. So I think it's good to have the defaults explicitly set, even if not necessary anymore. 
Source: https://docs.python.org/3/library/argparse.html#action